CONTRIBUTING.md

```diff
@@ -1,4 +1,81 @@
-# Developer's Certificate of Origin 1.1
+# Contributing to Undici
+
+* [Guides](#guides)
+  * [Update `llhttp`](#update-llhttp)
+* [Developer's Certificate of Origin 1.1](#developers-certificate-of-origin)
+  * [Moderation Policy](#moderation-policy)
+
+<a id="guides"></a>
+## Guides
+
+<a id="update-llhttp"></a>
+### Update `llhttp`
+
+The HTTP parser used by `undici` is a WebAssembly build of [`llhttp`](https://github.com/nodejs/llhttp).
+
+While the project itself provides a way to compile targeting WebAssembly, at the moment we embed the sources
+directly and compile the module in `undici`.
+
+The `deps/llhttp/include` folder contains the C header files, while the `deps/llhttp/src` folder contains
+the C source files needed to compile the module.
+
+The `lib/llhttp` folder contains the `.js` transpiled assets required to implement a parser.
+
+The following are the steps required to perform an update.
+
+#### Clone the [llhttp](https://github.com/nodejs/llhttp) project
+
+```bash
+git clone git@github.com:nodejs/llhttp.git
+
+cd llhttp
+```
+#### Checkout a `llhttp` release
+
+```bash
+git checkout <tag>
+```
+
+#### Install the `llhttp` dependencies
+
+```bash
+npm i
+```
+
+#### Run the wasm build script
+
+> This requires [docker](https://www.docker.com/) installed on your machine.
+
+```bash
+npm run build-wasm
+```
+
+#### Copy the sources to `undici`
+
+```bash
+cp build/wasm/*.js <your-path-to-undici>/lib/llhttp/
+
+cp build/wasm/*.js.map <your-path-to-undici>/lib/llhttp/
+
+cp build/wasm/*.d.ts <your-path-to-undici>/lib/llhttp/
+
+cp src/native/api.c src/native/http.c build/c/llhttp.c <your-path-to-undici>/deps/llhttp/src/
+
+cp src/native/api.h build/llhttp.h <your-path-to-undici>/deps/llhttp/include/
+```
+
+#### Build the WebAssembly module in `undici`
```

Comment:
Also this would not be needed of we didn't embed.

Reply:
Can we avoid embedding as of now?

Reply:
Yes, we can. We don't currently modify the source files. IMHO is a cleaner approach, as long as we don't have to customize it. One thing to consider is the direction the `llhttp` project will take towards wasm. For example, what if the only way to access the build would be by installing it as a dependency and use its `.js` API (as suggested in https://github.com/nodejs/llhttp/pull/93)? Would we be open to that?

But regarding now, we would just have to run `build-wasm` in `llhttp` and copy the `build/wasm` folder contents in `lib/llhttp`.